pkg/edgeview/src/system.go

```diff
@@ -1306,6 +1315,140 @@ func runTechSupport(cmds cmdOpt, isLocal bool) {
 	}
 }
 
+// edgeview request debug container to generate collect-info and
+// later to remove the tar.gz file
+// the protocol is for edgeview to create a file
+// /run/edgeview/edgeview-request-collect-info, and debug container
+// run 'collect-info.sh' to generate the tar.gz file, when the job is
+// done, remove the file /run/edgeview/edgeview-request-collect-info.
+// edgeview can also request to remove the tar.gz file by creating
+// /run/edgeview/edgeview-request-remove-tar.gz file, and debug container
+// will remove the generated tar.gz file.
+func runCollectInfo() {
```

Comment:
Isn't this over-complicated?
Instead of just calling `collect-info.sh` directly, you let a shell script in the debug container check if some files exist and then let it run. What are the advantages of doing it like this? I see a lot of code just for handling this re-invention of an RPC.

Reply:
i think the main thing is the avoid to expend the edgeview into similar capabilities of 'debug' container, has the 'eve enter' capability, etc. i can not do simply by using socket those things also; thus a simple file creation protocol for this specific task for the job.